### PR TITLE
feat(feature-flag): handle deployment date correctly

### PR DESCRIPTION
**Pull request by nicolasSagon**

### CODE REVIEW

**Comment by tristanrobert** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/mtes-mct/pr-checks/3e8bac22-f13e-4601-bf4a-ef2af69610c2?refs=5f239f74d6c4e4d9186cc5916fc8e6ce7b5790c0&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: dossierfacile-common-library/src/main/java/fr/dossierfacile/common/service/FeatureFlagServiceImpl.java [hunk 1]
```diff
@@ -83,15 +84,23 @@ private boolean checkAndAssign(Long userId, FeatureFlag featureFlag) {
...
-            userAssignmentBuilder.enabled(false)
-                    .bucket(0)
-                    .assignmentSource(FeatureAssignmentSource.PRE_DEPLOYMENT);
-
-            userAssignmentHistoryBuilder.enabled(false)
-                    .bucket(0);
-
-            return userFeatureAssignmentService.saveAssignment(userAssignmentBuilder.build(), userAssignmentHistoryBuilder.build(), false);
+        // We also test if the deployment date is not null at this moment.
+        if (featureFlag.isOnlyForNewUser() && featureFlag.getDeploymentDate() != null) {
```
Comment (alixdeschamps) [inline dossierfacile-common-library/src/main/java/fr/dossierfacile/common/service/FeatureFlagServiceImpl.java]: tu pourrais simplifier tes if else, là tu fais `if (featureFlag.isOnlyForNewUser() && ...` et plus bas `else if (featureFlag.isOnlyForNewUser()) {` . Extrait le `if (featureFlag.isOnlyForNewUser() `

**Review by alixdeschamps — APPROVED** [review]

**Review by Rhmaric — APPROVED** [review]
